extensions/itunes/src/album.tsx

```diff
@@ -0,0 +1,253 @@
+     -e "play playlist \\"ExtensionAlbumPlaying\\"" \
+     -e "end tell"`);
+      popToRoot().then();
+      closeMainWindow({ clearRootSearch: true }).then();
+    } catch (error) {
+      console.error("execution error", error);
+      showToast(
+        ToastStyle.Failure,
+        "Failed to play the track",
+        "Maybe the song is not in you library or has weird title"
+      ).then();
+      return;
+    }
+  });
+  return <Detail markdown={`# Calling ${using}...`} />;
+};
+
+export default function Command() {
+  const { state, search } = useSearch();
+
+  return (
+    <List
+      isLoading={state.isLoading}
+      onSearchTextChange={search}
+      searchBarPlaceholder="Search by name..."
+      throttle
+    >
+      <List.Section title="Results" subtitle={state.results.length + ""}>
+        {state.results.map((searchResult) => (
+          <SearchListItem key={searchResult.id} searchResult={searchResult} />
+        ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({ searchResult }: { searchResult: SearchResult }) {
+  return (
+    <List.Item
+      title={searchResult.album}
+      subtitle={searchResult.artist}
+      actions={
+        <ActionPanel>
+          <PushAction
+            title={`Play the Album`}
+            target={
+              <RunScript
+                album={searchResult.album}
+                artist={searchResult.artist}
+              />
+            }
+            icon={Icon.ArrowRight}
+          />
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+function useSearch() {
+  const [state, setState] = useState<SearchState>({
+    results: [],
+    isLoading: true,
+  });
+  const cancelRef = useRef<AbortController | null>(null);
+
+  useEffect(() => {
+    search("");
+    return () => {
+      cancelRef.current?.abort();
+    };
+  }, []);
+
+  async function search(searchText: string) {
+    cancelRef.current?.abort();
+    cancelRef.current = new AbortController();
+    try {
+      setState((oldState) => ({
+        ...oldState,
+        isLoading: true,
+      }));
+      await performSearch(searchText, cancelRef.current.signal);
+      setState((oldState) => ({
+        ...oldState,
+        results: match.map((entry) => {
+          return {
+            id: randomId(),
+            ...entry,
+          };
+        }),
+        isLoading: false,
+      }));
+    } catch (error) {
+      if (error instanceof AbortError) {
+        return;
+      }
+      console.error("search error", error);
+      showToast(
+        ToastStyle.Failure,
+        "Could not perform search",
+        String(error)
+      ).then();
+    }
+  }
+
+  return {
+    state: state,
+    search: search,
+  };
+}
+
+async function performSearch(searchText: string, signal: AbortSignal) {
+  if (searchText.length < 2) return;
+
+  let sherlock = false;
+  let dictBlock = false;
+  let buffer = {
+    album: "",
+    artist: "",
+  };
+  match = [];
+  const read = readline.createInterface({
```

Comment:
Do you think it might make sense to parse the XML once, get all the albums, and then let Raycast filter it? I don't quite understand the benefit of doing the filtering manually since you are parsing the entire XML file every time

Reply:
Because parsing the entire XML and storing then would blow the stack, my library XML contains over 270000 lines and that's normal.